### PR TITLE
New workflow: semgrep

### PR DESCRIPTION
**Pull request by thypon**
## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [x] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [x] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: .github/workflows/semgrep.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: "Semgrep"
+
+on:
+  pull_request:
```
Comment (diracdeltas) [inline .github/workflows/semgrep.yml:4]: maybe we should start by running these only against master PRs?

File: .github/workflows/semgrep.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    paths:
+      - .github/workflows/semgrep.yml
+  schedule:
+    - cron: '0 0 1 * *'
+
+jobs:
+  analyze:
+    name: Scan
+    runs-on: ubuntu-latest
+    if: (github.actor != 'dependabot[bot]')
```
Comment (diracdeltas) [inline .github/workflows/semgrep.yml:17]: can also exclude the renovate bot https://github.com/brave/brave-core/pulls/app%2Frenovate

**Review by diracdeltas — APPROVED** [review]

**Comment by bbondy** [comment]
Setting milestone to Nightly. Pls set that for future PRs that you merge. Thank you!
Also all things that are merged should have an issue associated with it.  Issues are what generate the release notes and also what QA builds test plans off of. The milestone would also be set on the issue if it existed.

**Comment by bbondy** [comment]
> [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue

That checklist item above btw is because it is a required thing for our process.
Thanks for your understanding!

**Pull request merged**